kernel/src/grant.rs

```diff
@@ -3,29 +3,26 @@
 use core::marker::PhantomData;
 use core::mem::{align_of, size_of};
 use core::ops::{Deref, DerefMut};
-use core::ptr::{slice_from_raw_parts_mut, write, NonNull};
+use core::ptr::{write, NonNull};
 
-use crate::process::{Error, ProcessType};
+use crate::process::{Error, ProcessDynamicGrantIdentifer, ProcessType};
 use crate::sched::Kernel;
 use crate::upcall::AppId;
 
-/// Type that indicates a grant region has been entered and borrowed.
-/// This is passed to capsules when they try to enter a grant region.
+/// This type provides access to the memory in a process's grant.
+///
+/// The grant must be entered to retrieve a `Borrowed` object. This object is
+/// what capsules are provided so they can access the grant memory.
 pub struct Borrowed<'a, T: 'a + ?Sized> {
+    /// The mutable reference to the actual object type stored in the grant.
     data: &'a mut T,
-    appid: AppId,
 }
 
 impl<'a, T: 'a + ?Sized> Borrowed<'a, T> {
-    fn new(data: &'a mut T, appid: AppId) -> Borrowed<'a, T> {
-        Borrowed {
-            data: data,
-            appid: appid,
-        }
-    }
-
-    pub fn appid(&self) -> AppId {
-        self.appid
+    /// Create a `Borrowed` object to provide to a capsule. Only one can be
```

Comment:
Not necessarily only to a capsule? Could it be that kernel services create grants too?